### ISSUE TITLE
RTL backup

### ISSUE THREAD

**Issue by frennkie** [raspiblitz/raspiblitz#1570]
I'm getting an error when trying to down a backup from RTL (`Lightning` > `Backup`).

RTL version 0.8.4-beta

Reading File Failed!
Error Code: -2
Error Message: ENOENT
API URL: ./api/conf/file?channel=ALL

![image](https://user-images.githubusercontent.com/6499251/93711069-65f69800-fb4b-11ea-839c-ef142d3ad51b.png)

**Comment by rootzoll** [comment]
I can reproduce this bug. Looking at the RTL logs it says:
```
Sep 21 15:00:05 raspberrypi node[17746]: [1B blob data]
Sep 21 15:00:05 raspberrypi node[17746]: ERROR: Conf(207) => Reading File Failed!
```

This error gets thrown by RTL here:
https://github.com/Ride-The-Lightning/RTL/blob/1b86a8bc84fae5708ec364711fbe26b22507906b/controllers/RTLConf.js#L207

@saubyk how can we make sure reading the file correct?

**Comment by kravens** [comment]
I had the same error, still want to be able to download a backup though... will try a scp command instead.

**Comment by saubyk** [comment]
Pinging @shahanafarooqui

**Comment by ShahanaFarooqui** [comment]
Above error only explains that the file doesn’t exist at the location. Just take the backup first and it will download the file.

You need to click on the ‘Backup All’ button to take the backup first. Once that button is clicked, the backup file will be saved in the backup folder and the file will be available for download after that. 

We have updated the error message to make it clearer. @saubyk, this is the same UX issue which we discussed two days back. We can backup channels if the file doesn’t pre-exist then download. But, you were not in favor to include these two actions in a single click.

**Comment by rootzoll** [comment]
@ShahanaFarooqui I just checked ... when I first click on "Backup All" I get a "all channels backuped sucessfully" but when I then click the "Download Backup" I still get the error described above. 

I checked ... the files gets created and is readable for all users:
![Bildschirmfoto 2020-09-28 um 11 55 02](https://user-images.githubusercontent.com/2472225/94418601-e63b7f80-0181-11eb-905d-7c17c804122b.png)

Tested on: 0.9.0-beta

**Comment by Marcoblue** [comment]
@ShahanaFarooqui I also clicked first "Backup all". Still the same error. Maybe, because it runs never before here?
With Thunderhub the backup is still working.

**Comment by NilByte** [comment]
Restore looks for backup file in:

Restore folder location: /home/rtl/RTL-SCB-backup-<ALIAS>/restore

Files are located in: /home/rtl/RTL-SCB-backup-<ALIAS>

**Comment by ShahanaFarooqui** [comment]
This bug fix will be released with next RTL version (https://github.com/Ride-The-Lightning/RTL/issues/536). It was due to case sensitive Unix file system. It is searching for file as ‘channel-ALL’ and unable to find the file. 

@NilByte The restore feature requires the user to place file in the ‘restore’ folder but this bug is about downloading the backup file on the local machine.

**Comment by rootzoll** [comment]
Check after installing the RTL update - see #2019

**Comment by rootzoll** [comment]
OK tested the backup again on new RTL 0.10.1 release included in v1.7RC1 --> not getting a error, but my backupfile is just 3 bytes @ShahanaFarooqui is there anything further to be changed/fixed?

**Comment by rootzoll** [comment]
In my latest tests in preparation of the v1.7.1 of RaspiBlitz I see the backup now also on the local directory with 0 bytes:
![Bildschirmfoto 2021-09-20 um 14 39 43](https://user-images.githubusercontent.com/2472225/134003880-4015624f-34c9-4a00-af81-028176750266.png)

**Comment by ShahanaFarooqui** [comment]
@rootzoll The issue has been fixed with release 10.0 and I am unable to reproduce the issue anymore. Can you please check if you are getting any error in RTL console related to channel backup?

**Comment by openoms** [comment]
Reproduces this on a running node, new RTL install 0.11.2. 
0 byte file was present from start. 
Logged in the interface and it even offered me to download a 0 byte one.
![image](https://user-images.githubusercontent.com/43343391/134081455-dcb4c02a-edc2-4574-92dd-75b944b91710.png)


The clicked on the `Backup All` after which a  48kbyte file appeared.
The logs show nothing unfortunately.
![image](https://user-images.githubusercontent.com/43343391/134081716-1e36d5cb-4213-46ff-9d54-cb198586c41c.png)



Can be reproduced if I delete the file and restart RTL, only a 0 byte one appears again.

**Comment by ShahanaFarooqui** [comment]
@openoms Thanks a lot for detailed testing and instructions. I was finally able to reproduce and fix the issue with your help :). 

The bug has been fixed and will be released with RTL v0.12.0.
